avalon/tools/loader/model.py

```diff
@@ -726,3 +815,142 @@ def sort(self, column, order):
             self.setSortRole(model.SortDescendingRole)
 
         super(FamiliesFilterProxyModel, self).sort(column, order)
+
+
+class RepresentationModel(TreeModel):
+
+    doc_fetched = QtCore.Signal()
+    refreshed = QtCore.Signal(bool)
+
+    Columns = [
+        "name",
+        "subset",
+        "asset",
+        "active_site",
+        "remote_site"
+    ]
+
+    def __init__(self, dbconn, header, version_ids):
+        super(RepresentationModel, self).__init__()
+        self.dbcon = dbconn
+        self._data = []
+        self._header = header
+        self.version_ids = version_ids
+
+        self.active_site = 'studio'  # TODO
+        self.remote_site = 'gdrive'  # TODO
+
+        self.doc_fetched.connect(self.on_doc_fetched)
+
+        self._docs = {}
+        self.repre_icons = get_repre_icons()
+
+    def set_version_ids(self, version_ids):
+        self.version_ids = version_ids
+        self.refresh()
+
+    def data(self, index, role):
+        if role == QtCore.Qt.DecorationRole:
+            if index.column() == self.Columns.index("active_site"):
+                item = index.internalPointer()
+                return item.get("active_site", None)
+            if index.column() == self.Columns.index("remote_site"):
+                item = index.internalPointer()
+                return item.get("remote_site", None)
+
+        return super(RepresentationModel, self).data(index, role)
+
+    def on_doc_fetched(self):
+        self.clear()
+        self.beginResetModel()
+        subsets = set()
+        assets = set()
+        for doc in self._docs:
+            progress = self._get_progress_for_repre(doc)
+
+            data = {
+                "name": doc["name"],
+                "subset": doc["context"]["subset"],
+                "asset": doc["context"]["asset"],
+
+                "active_site": self.repre_icons.get(self.active_site),
+                "remote_site": self.repre_icons.get(self.remote_site)
+            }
+            subsets.add(doc["context"]["subset"])
+            assets.add(doc["context"]["subset"])
+
+            item = Item()
+            item.update(data)
+
+            # TreeModel expects dict, attr here for later refactor
+            self.add_child(item, None)  # TODO parent
+
+        self.endResetModel()
+        self.refreshed.emit(False)
+
+    def refresh(self):
+
+        if self.version_ids:
+            # Simple find here for now, expected to receive lower number of
+            # representations and logic could be in Python
+            self._docs = self.dbcon.find(
+                {"type": "representation", "parent": {"$in": self.version_ids}}
+                , self._get_projection()
+            )
+
+        self.doc_fetched.emit()
+
+    def _get_projection(self):
```

Comment:
Rather use class attribute for projections then method. This way the same dictionary is created again and again.